doc/classes/Array.xml

```diff
@@ -142,7 +142,28 @@
 			<argument index="3" name="before" type="bool" default="true">
 			</argument>
 			<description>
-				Finds the index of an existing value (or the insertion index that maintains sorting order, if the value is not yet present in the array) using binary search and a custom comparison method. Optionally, a [code]before[/code] specifier can be passed. If [code]false[/code], the returned index comes after all existing entries of the value in the array. The custom method receives two arguments (an element from the array and the value searched for) and must return [code]true[/code] if the first argument is less than the second, and return [code]false[/code] otherwise.
+				Finds the index of an existing value (or the insertion index that maintains sorting order, if the value is not yet present in the array) using binary search and a custom comparison method declared in the obj. Optionally, a [code]before[/code] specifier can be passed. If [code]false[/code], the returned index comes after all existing entries of the value in the array. The custom method receives two arguments (an element from the array and the value searched for) and must return [code]true[/code] if the first argument is less than the second, and return [code]false[/code] otherwise.
+				[codeblock]
+				class custom:
+				    var val = 0
+				    func setV(a):
+				        val = a
+				    func getV():
+				        return val
+				    func _to_string():
+				        return String(val)
+				func compare(a,b): ## This compare is called when obj is set to self
+				    return a.getV() &lt; b.getV()
+				func _ready():
+				    var v = custom.new() ## Search value
+				    v.setV(8)
+				    var a = [custom.new(), custom.new(), custom.new(), custom.new(), custom.new(),
+				        custom.new(), custom.new(), custom.new(), custom.new(), custom.new(),
+				        custom.new(), custom.new(), custom.new(), custom.new(), custom.new()]
+				    for i in a.size():
+				        a[i].setV(i + 1)
+				    print(a.bsearch_custom(v, self, "compare", true)) #Expected value is 7
+				[/codeblock]
```

Comment:
```suggestion
				[codeblock]
				class Custom:
				    var val = 0

				    func set_v(a):
				        val = a

				    func get_v():
				        return val

				    func _to_string():
				        return String(val)

				func compare(a, b): # This compare is called when obj is set to self
				    return a.get_v() &lt; b.get_v()

				func _ready():
				    var v = custom.new() # Search value
				    v.set_v(8)
				    var a = [custom.new(), custom.new(), custom.new(), custom.new(), custom.new(),
				        custom.new(), custom.new(), custom.new(), custom.new(), custom.new(),
				        custom.new(), custom.new(), custom.new(), custom.new(), custom.new()]
				    for i in a.size():
				        a[i].set_v(i + 1)
				    print(a.bsearch_custom(v, self, "compare", true)) # Expected value is 7
				[/codeblock]
```

Reply:
What's the use of the `Custom` object in this example though? You could use a pre-existing type like String or even int to show how one can call a custom search method.

Reply:
I wasn't think much about it. You can use any object for this example. Ok, I will it change it to string.